lib/src/main/java/graphql/nadel/validation/NadelSchemaValidation.kt

```diff
@@ -29,10 +37,17 @@ class NadelSchemaValidation(
         val (engineSchema, services) = schemas
 
         val servicesByName = services.strictAssociateBy(Service::name)
+
+        val operationTypes = getOperationTypeNames(engineSchema)
+        val namespaceTypes = getNamespaceOperationTypes(engineSchema)
+
         val context = NadelValidationContext(
-            engineSchema,
-            servicesByName,
-            fieldContributorMap,
+            engineSchema = engineSchema,
+            servicesByName = servicesByName,
+            fieldContributor = fieldContributorMap,
+            hydrationUnions = getHydrationUnions(engineSchema),
+            namespaceTypeNames = namespaceTypes,
+            combinedTypeNames = namespaceTypes + operationTypes.map { it.name },
```

Comment:
Store some more precomputed info like `hydrationUnions` , `namespaceTypeNames` and `combinedTypeNames`